src/App.js

```diff
@@ -1,84 +1,57 @@
-import React from "react";
-import "./App.css";
+import React, { useState, useEffect } from 'react';
+import './App.css';
+import CurrentWeather from './components/currentWeather';
+import WeatherForecast from './components/WeatherForecast';
+import { filterForecast } from './utils/functions';
+import SelectCity from './components/selectCity';
+
+const myId = process.env.REACT_APP_MY_API_ID;
 
 const App = () => {
+  const [weather, setWeather] = useState(null);
+  const [city, setCity] = useState('Prague');
+  const [forecast, setForecast] = useState(null);
+
+  const fetchWeather = (city) => {
+    fetch(
+      `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&APPID=${myId}`,
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        setWeather(data);
+      });
+  };
+
+  const fetchForecast = (city) => {
+    fetch(
+      `https://api.openweathermap.org/data/2.5/forecast?q=${city}&units=metric&APPID=${myId}`,
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        setForecast(filterForecast(data.list));
+      });
+  };
+
+  const handleChangeCity = (c) => {
+    setCity(c);
+  };
+
+  useEffect(() => {
+    fetchWeather(city);
+    fetchForecast(city);
+  }, [city]);
```

Comment:
Tohle je dobrý, ale je potřeba si uvědomit že tyhle 2 operace běží paralelně. Tj kdyby např fetchWeather trval 3s (může se někde seknout), a fetchForecast 500ms, tak se ti UI updatuje "po kouskách". Občas to chceš, ale většinou načítání dat poslouchá tenhle pattern:
- nastav loading stav
- načti z API všechna data která potřebuješ
- nastav potřebný stav a vypni loading

Plus ještě chceš ošetřit případ, kdy někdo rychle přepíná města, rychlejc než se stačí načíst. Chceš ty staré výsledky co dojdou později než nové zahodit;

tj v tomhle případě:
```
useEffect(() => {
   let cancelled = false;

   setWeather(null);
   setForecast(null);

   Promise.all([
      fetch(`https://api.openweathermap.org/data/2.5/weather?appid=${API_KEY}&q=${city}&units=metric`).then((response) => response.json()),
fetch(`https://api.openweathermap.org/data/2.5/forecast?appid=${API_KEY}&q=${city}&units=metric`)
    .then((response) => response.json()),
   ]).then(([weather, forecast]) => {
        if(!cancelled){
            setWeather(weather);
            setForecast(filterForecast(data.list));
        }
   });

   return () => {
      cancelled = true;
   }
}, [city])
```
Ale to ti říkám jen pro informaci a abys viděla, že jakékoli programování s nějakými asynchroními věcmi začíná být o dost složitější :)

Reply:
super, dekuji!